daemon/graphdriver/devmapper/driver.go

```diff
@@ -151,7 +151,7 @@ func (d *Driver) Remove(id string) error {
 
 	// This assumes the device has been properly Get/Put:ed and thus is unmounted
 	if err := d.DeviceSet.DeleteDevice(id, false); err != nil {
-		return err
+		return fmt.Errorf("failed to remove device %v:%v", id, err)
```

Comment:
fixed https://github.com/docker/docker/issues/30001 (and added commit here as well)